motif/core.py

```diff
@@ -264,9 +346,99 @@ def save(self, output_fpath):
                 self.salience
             ))
 
+
+def _validate_contours(index, times, freqs, salience):
+    N = len(index)
+    if any([len(times) != N, len(freqs) != N, len(salience) != N]):
+        raise ValueError(
+            "the arrays index, times, freqs, and salience "
+            "must be the same length."
+        )
+
+
+def _format_contour_data(frequencies):
+    """ Convert contour frequencies to cents + voicing.
+
+    Parameters
+    ----------
+    frequencies : np.array
+        Contour frequency values
+
+    Returns
+    -------
+    est_cent : np.array
+        Contour frequencies in cents
+    est_voicing : np.array
+        Contour voicings
+
+    """
+    est_freqs, est_voicing = mir_eval.melody.freq_to_voicing(frequencies)
+    est_cents = mir_eval.melody.hz2cents(est_freqs, 10.)
+    return est_cents, est_voicing
+
+
+def _format_annotation(annot_times, annot_freqs, duration, sample_rate):
+    """ Format an annotation file and resample to a uniform timebase.
+
+    Parameters
+    ----------
+    annot_times : np.array
+        Annotation time stamps
+    annot_freqs : np.array
+        Annotation frequency values
+    duration : float
+        Length of the full audio file in seconds.
+    sample_rate : float
+        The target sample rate.
+
+    Returns
+    -------
+    annot_times_new : np.array
+        New annotation time stamps
+    ref_cent : np.array
+        Annotation frequencies in cents at the new timescale
+    ref_voicing : np.array
+        Annotation voicings at the new timescale
+
+    """
+    annot_times_new = np.arange(0, duration + 0.5/sample_rate, 1.0/sample_rate)
+
+    ref_freq, ref_voicing = mir_eval.melody.freq_to_voicing(annot_freqs)
+    ref_cent = mir_eval.melody.hz2cents(ref_freq, 10.)
+
+    ref_cent, ref_voicing = mir_eval.melody.resample_melody_series(
+        annot_times, ref_cent, ref_voicing, annot_times_new,
+        kind='linear'
+    )
+    return annot_times_new, ref_cent, ref_voicing
+
+
+def _get_snippet_idx(snippet, full_array):
```

Comment:
maybe I would move this function inside `utils.py`, since it's not using anything from the core classes